### PR TITLE
Change softfail into fail, add disk check after detaching

### PR DESCRIPTION
**Pull request by RoyCai7**
poo#[122617](https://progress.opensuse.org/issues/122617)
Because softfail can ignore issue, change it into fail. Add disk check after detching disk.

- Related ticket: [122617](https://progress.opensuse.org/issues/122617)
- Needles:N/A
- Verification run:  [KVM](http://openqa.qam.suse.cz/tests/51454#step/hotplugging_HDD/141)  [XEN](http://openqa.qam.suse.cz/tests/51469#step/hotplugging_HDD/207)

### CODE REVIEW

File: tests/virtualization/universal/hotplugging_HDD.pm [hunk 1]
```diff
@@ -38,17 +38,19 @@ sub test_add_virtual_disk {
...
-                record_info('Softfail', "lsblk failed - please check the output manually", result => 'softfail') if $lsblk != 0;
+                die("lsblk failed - please check the output manually") if $lsblk != 0;
             } elsif (is_xen_host) {
                 my $lsblk = script_run("ssh root\@$guest lsblk | grep 'xvd[b-z]'", 60);
-                record_info('Softfail', "lsblk failed - please check the output manually", result => 'softfail') if $lsblk != 0;
+                die("lsblk failed - please check the output manually") if $lsblk != 0;
             } else {
                 my $msg = "Unknown virtualization hosts";
-                record_info('Softfail', $msg, result => 'softfail');
+                die($msg);
```
Comment (waynechen55) [inline tests/virtualization/universal/hotplugging_HDD.pm:47]: One line is enough for all branches IMHO.

**Comment by RoyCai7** [comment]
@alice-suse @xguo @waynechen55 @Julie-CAO @tbaev @nanzhg @varunkojha please help to review this PR. Thanks

**Review by alice-suse — APPROVED** [review]
LGTM

**Review by waynechen55 — COMMENTED** [review]
If you change to die, why not using assert_script_run directly ?

**Comment by RoyCai7** [comment]
> If you change to die, why not using assert_script_run directly ?

Yes, it is best option.

**Comment by nanzhg** [comment]
> If you change to die, why not using assert_script_run directly ?

Agree. it will be more concise by using this subroutine. 
assert_script_run($cmd [, timeout => $timeout] [, fail_message => $fail_message] [,quiet => $quiet]);

File: tests/virtualization/universal/hotplugging_HDD.pm [hunk 2]
```diff
@@ -34,21 +34,12 @@ sub test_add_virtual_disk {
...
-                my $msg = "Unknown virtualization hosts";
-                record_info('Softfail', $msg, result => 'softfail');
-            }
-        }
         assert_script_run("ssh root\@$guest lsblk");
+        # Attach disk check
+        assert_script_run("ssh root\@$guest lsblk | grep -iE '[x]\?vd[b-z]'", timeout => 60, fail_message => "Failed to attach disk for guest $guest");
         assert_script_run("virsh detach-disk $guest ${domblk_target}", 240);
+        # Detach disk check
+        die "Failed to detach disk for guest $guest." if (script_run("ssh root\@$guest lsblk | grep -iE '[x]\?vd[b-z]'", 60) eq 0);
```
Comment (waynechen55) [inline tests/virtualization/universal/hotplugging_HDD.pm:42]:
```
assert_script_run("! ssh root\@$guest lsblk | grep -iE '[x]\?vd[b-z]'")
```
???

Comment (RoyCai7) [reply tests/virtualization/universal/hotplugging_HDD.pm]: It should be OK, I will try.

**Review by waynechen55 — APPROVED** [review]

**Pull request merged**